File: samples/VB .NET Samples/DotNETTWAINDemo/DotNETTWAINDemo/CameraUI.vb
Imports Dynamsoft.Core
Imports Dynamsoft.UVC
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms

Public Partial Class CameraUI
	Inherits Form
	Public Sub New()
		InitializeComponent()
	End Sub

	Private m_Camera As Camera = Nothing
	Public Property Camera() As Camera
		Get
			Return m_Camera
		End Get
		Set
			m_Camera = value
			m_Camera.SetVideoContainer(Me.Handle)
		End Set
	End Property


	Private Sub CameraUI_FormClosed(sender As Object, e As FormClosedEventArgs)
		If m_Camera IsNot Nothing Then
			m_Camera.Close()
		End If
	End Sub

	Private Sub CameraUI_SizeChanged(sender As Object, e As EventArgs)
		If m_Camera IsNot Nothing Then
			m_Camera.ResizeVideoWindow(0, menuStrip1.Height, Me.ClientRectangle.Width, Me.ClientRectangle.Height)
		End If

	End Sub

	Private Sub optionsToolStripMenuItem_Click(sender As Object, e As EventArgs)
		m_Camera.DisplayPropertyPage()
	End Sub
End Class

File: samples/C# Samples/DotNETTWAINDemo/DotNETTWAINDemo/CameraUI.cs
using Dynamsoft.Core;
using Dynamsoft.UVC;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;

namespace DotNet_TWAIN_Demo
{
    public partial class CameraUI : Form
    {
        public CameraUI()
        {
            InitializeComponent();
        }

        private Camera m_Camera = null; 
        public Camera Camera
        {
            set
            {
                m_Camera = value;
                m_Camera.SetVideoContainer(this.Handle);
            }
            get
            {
                return m_Camera;
            }
        }


        private void CameraUI_FormClosed(object sender, FormClosedEventArgs e)
        {
            if (m_Camera != null)
                m_Camera.Close();
        }

        private void CameraUI_SizeChanged(object sender, EventArgs e)
        {
            if (m_Camera!=null)
            {
                m_Camera.ResizeVideoWindow(0, menuStrip1.Height, this.ClientRectangle.Width, this.ClientRectangle.Height);
            }

        }

        private void optionsToolStripMenuItem_Click(object sender, EventArgs e)
        {
            m_Camera.DisplayPropertyPage();
        }
    }
}
